pkg/resourcekeeper/gc.go

```diff
@@ -245,23 +245,71 @@ func (h *gcHandler) Sweep(ctx context.Context) (finished bool, waiting []v1beta1
 }
 
 func (h *gcHandler) recycleResourceTracker(ctx context.Context, rt *v1beta1.ResourceTracker) error {
+	var (
+		pendingDelMRs []*v1beta1.ManagedResource
+		todoDelMRs    []*v1beta1.ManagedResource
+	)
+	var err error
 	for _, mr := range rt.Spec.ManagedResources {
-		entry := h.cache.get(ctx, mr)
-		if entry.gcExecutorRT != rt {
-			continue
-		}
-		if entry.err != nil {
-			return entry.err
+		todoDelMRs, pendingDelMRs, err = h.checkPending(ctx, mr, rt, todoDelMRs, pendingDelMRs)
+		if err != nil {
+			return err
 		}
-		if entry.exists {
-			if err := h.Client.Delete(multicluster.ContextWithClusterName(ctx, mr.Cluster), entry.obj); err != nil && !kerrors.IsNotFound(err) {
-				return errors.Wrapf(err, "failed to delete resource %s", mr.ResourceKey())
+	}
+	if len(todoDelMRs) > 0 {
+		for _, mr := range todoDelMRs {
+			entry := h.cache.get(ctx, *mr)
+			if entry.gcExecutorRT != rt {
+				continue
 			}
+			if entry.err != nil {
+				return entry.err
+			}
+			if entry.exists {
+				if err := h.Client.Delete(multicluster.ContextWithClusterName(ctx, mr.Cluster), entry.obj); err != nil && !kerrors.IsNotFound(err) {
+					return errors.Wrapf(err, "failed to delete resource %s", mr.ResourceKey())
+				}
+				h.cache.setNotExist(*mr)
+			}
+		}
+		if len(todoDelMRs) == 0 && len(pendingDelMRs) == 0 {
+			return nil
+		}
+		if len(pendingDelMRs) > 0 {
+			return h.recycleResourceTracker(ctx, rt)
 		}
 	}
 	return nil
 }
 
+func (h *gcHandler) checkPending(ctx context.Context, mr v1beta1.ManagedResource, rt *v1beta1.ResourceTracker, todoDelMRs []*v1beta1.ManagedResource, pendingDelMRs []*v1beta1.ManagedResource) ([]*v1beta1.ManagedResource, []*v1beta1.ManagedResource, error) {
+	// if the ManagedResource is not exist so return directly
```

Comment:
if the ManagedResource does not exist, then return directly

Reply:
yes, the ManagedResource does not exist means that gc for the resource is done

Reply:
I mean the comment has some grammar error :)